src/inc_gauged_raz.rs

```diff
@@ -37,6 +37,7 @@ const DEFAULT_SECTION_CAPACITY: usize = 500;
 /// The data stored in the tree structure of the RAZ.
 #[derive(PartialEq,Eq,Debug,Hash,Clone)]
 enum TreeData<E:Debug+Clone+Eq+Hash, M:RazMeta<E>> {
+	Dummy, // used when rebuild is certain
```

Comment:
Rebuilds only occur on branches, but now branches need to construct meta data, and I didn't like cloning names only to have them thrown out. `Leaf(Rc::new(vec![]))` could work, but it might confuse algorithms that use the convention that leaves alway hold some data and never have branches. I think an explicit type indicating purpose makes everything clearer. As a bonus it avoids constructing any unnecessary data.